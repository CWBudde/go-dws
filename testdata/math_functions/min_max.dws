// Test Min() and Max() functions
// Task 9.65: Min/Max test fixture
// Min(a, b) returns the smaller value
// Max(a, b) returns the larger value

begin
	// Test Min with integers
	PrintLn('Min(5, 10) = ', Min(5, 10));
	PrintLn('Min(10, 5) = ', Min(10, 5));
	PrintLn('Min(-5, -10) = ', Min(-5, -10));
	PrintLn('Min(-5, 10) = ', Min(-5, 10));
	PrintLn('Min(7, 7) = ', Min(7, 7));
	PrintLn('Min(0, 5) = ', Min(0, 5));

	// Test Max with integers
	PrintLn('Max(5, 10) = ', Max(5, 10));
	PrintLn('Max(10, 5) = ', Max(10, 5));
	PrintLn('Max(-5, -10) = ', Max(-5, -10));
	PrintLn('Max(-5, 10) = ', Max(-5, 10));
	PrintLn('Max(7, 7) = ', Max(7, 7));
	PrintLn('Max(0, -5) = ', Max(0, -5));

	// Test Min with floats
	PrintLn('Min(3.14, 2.71) = ', Min(3.14, 2.71));
	PrintLn('Min(2.71, 3.14) = ', Min(2.71, 3.14));
	PrintLn('Min(-1.5, -2.5) = ', Min(-1.5, -2.5));
	PrintLn('Min(1.5, 1.5) = ', Min(1.5, 1.5));

	// Test Max with floats
	PrintLn('Max(3.14, 2.71) = ', Max(3.14, 2.71));
	PrintLn('Max(2.71, 3.14) = ', Max(2.71, 3.14));
	PrintLn('Max(-1.5, -2.5) = ', Max(-1.5, -2.5));
	PrintLn('Max(1.5, 1.5) = ', Max(1.5, 1.5));

	// Test Min/Max with mixed types (Integer and Float)
	PrintLn('Min(5, 3.14) = ', Min(5, 3.14));
	PrintLn('Min(3.14, 5) = ', Min(3.14, 5));
	PrintLn('Max(5, 3.14) = ', Max(5, 3.14));
	PrintLn('Max(3.14, 5) = ', Max(3.14, 5));

	// Test in expressions
	var x := Min(100, 50) + Max(10, 20);
	PrintLn('Min(100, 50) + Max(10, 20) = ', x);
end
