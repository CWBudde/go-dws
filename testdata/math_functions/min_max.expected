Min(5, 10) =  5
Min(10, 5) =  5
Min(-5, -10) =  -10
Min(-5, 10) =  -5
Min(7, 7) =  7
Min(0, 5) =  0
Max(5, 10) =  10
Max(10, 5) =  10
Max(-5, -10) =  -5
Max(-5, 10) =  10
Max(7, 7) =  7
Max(0, -5) =  0
Min(3.14, 2.71) =  2.71
Min(2.71, 3.14) =  2.71
Min(-1.5, -2.5) =  -2.5
Min(1.5, 1.5) =  1.5
Max(3.14, 2.71) =  3.14
Max(2.71, 3.14) =  3.14
Max(-1.5, -2.5) =  -1.5
Max(1.5, 1.5) =  1.5
Min(5, 3.14) =  3.14
Min(3.14, 5) =  3.14
Max(5, 3.14) =  5
Max(3.14, 5) =  5
Min(100, 50) + Max(10, 20) =  70
